plugins/kubernetes/src/components/Pods/PodDrawer/ContainerCard.tsx

```diff
@@ -222,13 +221,19 @@ export const ContainerCard: React.FC<ContainerCardProps> = ({
           )}
         </Grid>
       </CardContent>
-      <CardActions disableSpacing>
+      <CardActions>
         <PodLogsDialog
           containerScope={{
             containerName: containerStatus.name,
             ...podScope,
           }}
         />
+        <K8sTerminalDialog
+          clusterName={podScope.clusterName}
+          containerName={containerStatus.name}
+          podName={podScope.podName}
+          podNamespace={podScope.podNamespace}
+        />
```

Comment:
Can we disable this button in cases where we know the exec API won't work?

The way we have built our backend means that this dialog will only be usable if:
1. you have a single cluster, and
2. that cluster makes use of a server-side auth provider.

If you have multiple clusters, then the WebSocket handshake will respond with 404 and body
```json
{
  "error": {
    "name": "NotFoundError",
    "message": "Cluster 'undefined' not found",
    "stack": "NotFoundError: Cluster 'undefined' not found\n    at KubernetesProxy.getClusterForRequest (webpack-internal:///../../plugins/kubernetes-backend/src/service/KubernetesProxy.ts:164:19)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async eval (webpack-internal:///../../plugins/kubernetes-backend/src/service/KubernetesProxy.ts:86:49)"
  },
  "request": {
    "method": "GET",
    "url": "/api/kubernetes/proxy/api/v1/namespaces/:namespace/pods/:podName/exec?container=nginx&stdin=true&stdout=true&stderr=true&tty=true&command=%2Fbin%2Fsh"
  },
  "response": {
    "statusCode": 404
  }
}
```

and if you have a single cluster with a client-side auth provider (say, `aks`), it will 401 with body

```json
{
  "kind": "Status",
  "apiVersion": "v1",
  "metadata": {},
  "status": "Failure",
  "message": "Unauthorized",
  "reason": "Unauthorized",
  "code": 401
}
```

Reply:
From our in-person conversation: If you import [`kubernetesApiRef`](https://backstage.io/docs/reference/plugin-kubernetes.kubernetesapiref/) you can do a `useApi(kubernetesApiRef)` and the thing you get will have a [`getClusters()` method](https://backstage.io/docs/reference/plugin-kubernetes.kubernetesapi.getclusters)

Reply:
Added cluster size 1 check, what about the auth 404?

Reply:
Your check handles the 404 case, the only problem now is if your single cluster requires a [client-side credential](https://backstage.io/docs/features/kubernetes/authentication#client-side-providers). It's not the most maintainable, but you could check that the `authProvider` does not match `aks`, `google`, or `oidc.*`

Reply:
Are we blacklisting those 3 or whitelisting the others? which ones are valid then?

Reply:
the button should be disabled when the authprovider is one of those 3 -- those are the client-side auth providers and we have no way for the client to provide a token since the WebSocket API doesn't allow custom headers

Reply:
Some tests broke when importing the kubernetes api, can't find any examples of it being mocked in UI tests, and the `MockKubernetesClient` in `plugins/kubernetes/dev/index.tsx` isn't even exported, the closest example I can find is `plugins/kubernetes/src/hooks/useKubernetesObjects.test.ts` in which the tests are not providing the mock through `TestApiProvider`, but instead is mocking the module's `useApi` to return the mock instead, kind of confused which of these approaches I should be using...

Reply:
Done